src/gretel_trainer/trainer.py

```diff
@@ -58,43 +40,56 @@ class Trainer:
     def __init__(
         self,
         project_name: str = "trainer",
-        max_header_clusters: int = 20,
-        max_rows: int = 50000,
+        max_header_clusters: int = None,
+        max_rows: int = None,
         model_type: str = "GretelLSTM",
         model_params: dict = {},
         cache_file: str = None,
         overwrite: bool = True,
         enable_privacy_filters: bool = False,
     ):
-
         configure_session(api_key="prompt", cache="yes", validate=True)
 
         self.df = None
         self.dataset_path = None
+        self.max_header_clusters = max_header_clusters
+        self.max_rows = max_rows
         self.run = None
         self.project_name = project_name
         self.project = create_or_get_unique_project(name=project_name)
-        self.max_header_clusters = max_header_clusters
-        self.max_rows = max_rows
         self.overwrite = overwrite
         self.cache_file = self._get_cache_file(cache_file)
 
-        if model_type in Model.get_types():
-            self.config = read_model_config(Model.get_config(model_type))
+        # Verify model specific settings
+        if model_type not in Model.get_types():
```

Comment:
If we used a `model_config: BaseConfig` here, you could just do an `isinstance(model_config, BaseConfig)` to ensure we have a valid config